obelisk/cpp/obelisk_cpp/include/obelisk_mujoco_sim_robot.h

```diff
@@ -0,0 +1,635 @@
+
+            // Read in the config string
+            std::string mujoco_setting = this->get_parameter("mujoco_setting").as_string();
+            auto mujoco_config_map     = this->ParseConfigStr(mujoco_setting);
+
+            // Get config params
+            xml_path_ = GetXMLPath(mujoco_config_map); // Required
+            // Search for the model
+            if (!std::filesystem::exists(xml_path_)) {
+                if (const char* obelisk_root = std::getenv("OBELISK_ROOT")) {
+                    xml_path_ = "models" / xml_path_;
+                    xml_path_ = static_cast<std::string>(obelisk_root) / xml_path_;
+                } else {
+                    throw std::runtime_error("OBELISK_ROOT environment variable not set. Run the dev_setup.sh script!");
+                }
+            }
+
+            nu_                = GetNumInputs(mujoco_config_map); // Required
+            time_step_         = GetTimeSteps(mujoco_config_map);
+            num_steps_per_viz_ = GetNumStepsPerViz(mujoco_config_map);
+
+            shared_data_.resize(nu_);
+
+            configuration_complete_ = true;
+
+            // Setup the sensors
+            try {
+                ParseSensorString(mujoco_config_map.at("sensor_settings"));
+            } catch (const std::exception& e) {
+                RCLCPP_WARN_STREAM(this->get_logger(), "Mujoco simulation initialized without any sensors.");
+            }
+
+            return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
+        }
+
+        /**
+         * @brief activates up the node.
+         *
+         * @param prev_state the state of the ros node.
+         */
+        rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn virtual on_activate(
+            const rclcpp_lifecycle::State& prev_state) {
+            this->ObeliskSimRobot<ControlMessageT>::on_activate(prev_state);
+
+            return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
+        }
+
+        /**
+         * @brief deactivates up the node.
+         *
+         * @param prev_state the state of the ros node.
+         */
+        rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn virtual on_deactivate(
+            const rclcpp_lifecycle::State& prev_state) {
+            this->ObeliskSimRobot<ControlMessageT>::on_deactivate(prev_state);
+
+            return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
+        }
+
+        /**
+         * @brief cleans up the node.
+         *
+         * @param prev_state the state of the ros node.
+         */
+        rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn virtual on_cleanup(
+            const rclcpp_lifecycle::State& prev_state) {
+            this->ObeliskSimRobot<ControlMessageT>::on_cleanup(prev_state);
+            configuration_complete_ = false;
```

Comment:
what's the point of this variable?

Reply:
`configuration_complete_` is just a flag to ensure that the stuff in the sim thread does not attempt to access data before it has been created.